hrms/hr/doctype/leave_control_panel/leave_control_panel.js

```diff
@@ -2,23 +2,175 @@
 // License: GNU General Public License v3. See license.txt
 
 frappe.ui.form.on("Leave Control Panel", {
-	onload: function(frm) {
-		if (!frm.doc.from_date) {
-			frm.set_value('from_date', frappe.datetime.get_today());
-		}
-	},
-	refresh: function(frm) {
+	refresh: function (frm) {
 		frm.disable_save();
+		frm.trigger("reset_leave_details");
+		frm.trigger("load_employees");
+		frm.trigger("set_primary_action");
 	},
-	company: function(frm) {
-		if(frm.doc.company) {
-			frm.set_query("department", function() {
+	company: function (frm) {
+		if (frm.doc.company) {
+			frm.set_query("department", function () {
 				return {
-					"filters": {
-						"company": frm.doc.company,
-					}
+					filters: {
+						company: frm.doc.company,
+					},
 				};
 			});
 		}
-	}
+		frm.trigger("load_employees");
+	},
+	employment_type(frm) {
+		frm.trigger("load_employees");
+	},
+
+	branch(frm) {
+		frm.trigger("load_employees");
+	},
+
+	department(frm) {
+		frm.trigger("load_employees");
+	},
+
+	designation(frm) {
+		frm.trigger("load_employees");
+	},
+
+	employee_grade(frm) {
+		frm.trigger("load_employees");
+	},
+
+	reset_leave_details(frm) {
+		frm.set_value({
+			dates_based_on: undefined,
+			from_date: frappe.datetime.get_today(),
+			to_date: undefined,
+			leave_period: undefined,
+			carry_forward: 0,
+			allocation_based_on: undefined,
```

Comment:
The default value for `allocation_based_on` gets reset here

![image](https://github.com/frappe/hrms/assets/24353136/6aa0fc47-1261-49bf-9f74-856e88a4cc23)

Also instead of having a select field with a single selection, we should have a checkbox maybe:
**Allocate Based On Leave Policy**.
This should be checked by default.

Even in the dates selection, remove the empty option

Dates Based On:
1. Leave Period
2. Joining Date
3. Custom Date Range

Reply:
Added in https://github.com/frappe/hrms/pull/809/commits/bab3a3c56104a1836814f1b463951128d0e8d788